marketplace-service/src/main/java/com/axonivy/market/service/impl/ProductServiceImpl.java

```diff
@@ -574,4 +562,70 @@ public void transferComputedDataFromDB(Product product) {
     );
   }
 
+  @Override
+  public Product renewProductById(String id) {
+    Product product = new Product();
+    productRepository.findById(id).ifPresent(foundProduct -> {
+          ProductFactory.transferComputedPersistedDataToProduct(foundProduct, product);
+          imageRepository.deleteAllByProductId(foundProduct.getId());
+          productModuleContentRepository.deleteAllByProductId(foundProduct.getId());
+          productJsonContentRepository.deleteAllByProductId(foundProduct.getId());
+          productRepository.delete(foundProduct);
+        }
+    );
+
+    return product;
+  }
+
+  @Override
+  public boolean syncOneProduct(String marketItemPath, Product product) {
+    try {
+      var gitHubContents = axonIvyMarketRepoService.getMarketItemByPath(marketItemPath);
+      if (!CollectionUtils.isEmpty(gitHubContents)) {
+        mappingMetaDataAndLogoFromGHContent(gitHubContents, product);
+        updateRelatedThingsOfProductFromGHContent(gitHubContents, product);
+        productRepository.save(product);
+        return true;
+      }
+    } catch (Exception e) {
+      log.error(e.getStackTrace());
+    }
+    return false;
+  }
+
+  private void mappingMetaDataAndLogoFromGHContent(List<GHContent> gitHubContent, Product product) {
+    GHContent metaFile = null;
+    GHContent logoFile = null;
+    for (var content : gitHubContent) {
+      if (StringUtils.endsWith(content.getName(), META_FILE)) {
+        metaFile = content;
+      } else {
+        logoFile = content;
+      }
+    }
+
+    ProductFactory.mappingByGHContent(product, metaFile);
```

Comment:
please replace the line 607 to 601 , and replace line 608 to 603

Reply:
I want to sync the meta first. So I will keep it atm.